src/providers/EtherspotContextProvider.tsx

```diff
@@ -511,6 +512,43 @@ const EtherspotContextProvider = ({
     [sdk, accountAddress]
   );
 
+  // get ENS Node
+  const getEnsNode = useCallback(
+    async (
+      chainId: number,
+      address: string | null = accountAddress,
+      recompute: boolean = true
+    ): Promise<ENSNode | null> => {
+      const sdk = getSdkForChainId(chainId);
+
+      if (!sdk) return null;
+
+      let computedAccount;
+
+      if (!address && recompute) {
+        try {
+          computedAccount = await connect();
+        } catch (e) {
+          //
+        }
+      }
+
+      if (!address && !computedAccount) return null;
+
+      try {
+        const ens = await sdk.getENSNode({
+          nameOrHashOrAddress: address || computedAccount || '',
```

Comment:
that should help, empty string fallback is excess
```suggestion
          nameOrHashOrAddress: address ?? computedAccount
```